### PR TITLE
Remove `nuxt-ssr-cache` and move dependencies

### LINKED ISSUE #414: Move away from `nuxt-ssr-cache`
## Problem

For microcaching of our server-rendered pages, we currently use a poorly-maintained package `nuxt-ssr-cache`. We also use `patch-package` to modify this broken package.

## Solution

There are a few alternative solutions:

1. Switch to an NGINX based approach to microcaching: https://www.nginx.com/blog/benefits-of-microcaching-nginx/
2. Switch to nuxt-perfect-cache with Redis: https://www.npmjs.com/package/nuxt-perfect-cache

I prefer approach 1, as it removes caching from an app-level consideration and into our infra layer.

With either approach we can:

1. Remove the `"postinstall": "patch-package",` script
2. Remove the `patches/` directory

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #414 by @zackkrida
Related to #388 by @rbadillap 

## Description

Doing the following changes should prepare for the deployment process with Docker.
- Removes `nuxt-ssr-cache`and `patch-package`
- Removes the postinstall script and the patches folder
- Moves `sass-loader` and `sass` to devDependencies
- Moves `core-js`, `glob` and `vue-i18n-extract` to production dependencies

## Testing Instructions

1. Run `npm run install && npm run dev`
2. Check the site at `http://localhost:8443/`
3. ...

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: package.json
Comment (obulat) [inline package.json:69]: This is the package for extracting any untranslated strings from the app, and not needed in prod, so it should be a dev dependency.

**Comment by obulat** [comment]
This PR is not supposed to get the Docker build running, right, @krysal ? Because I get an error saying that it doesn't have Python for `node-gyp`.
Are we going to have no caching until we create the new caching solution?

File: package.json
Comment (krysal) [reply package.json:69]: It is not needed for the `i18n:get-translations` script? Because I think the intention is to be able to run that command in production as well.

**Comment by krysal** [comment]
@obulat No, this doesn't address the Docker build, this moves the aforementioned dependencies to where they correspond according to the project requirements. I should have worded it better above, it just seems to me this needs to be resolved so @rbadillap can move on with Docker.

About the micro caching, yes, as @zackkrida mentioned, the [nuxt-ssr-cache](https://github.com/arash16/nuxt-ssr-cache) package is broken and has not received an update in over a year, the other is too new and seems to be maintained only by one person, so I think the NGINX option is better too.

**Comment by zackkrida** [comment]
The only thing I'm still thinking about is that we _might_ want to wait until after launch to merge this and before retooling the front-end deployment process in general.

File: package.json
Comment (obulat) [reply package.json:69]: Sorry I haven't checked for replies, @krysal . No, `vue-i18n-extract` shouldn't be used. I did copy some parts of it in `parse-vue-files` 😁

**Comment by zackkrida** [comment]
@krysal do you know why all these formatting fixes are included in this branch? I'm really excited to move forward with this (I also repointed it to a new `docker-improvements` branch) but those formatting fixes make it harder to test.

**Comment by zackkrida** [comment]
Ohh it was because prettier updated! Got it 👍

**Comment by krysal** [comment]
@zackkrida Yes, sorry for the formatting. The good thing is that I discovered [the issue](https://github.com/vuejs/vetur/issues/3156#issuecomment-924126544) with VS Code formatting the code like this every time when saving a file, and now is fixed :)

File: package.json
Comment (krysal) [reply package.json]: Moved in [`1875f10`](https://github.com/WordPress/openverse-frontend/pull/415/commits/1875f10a18190cb005559f8296f5c061dbef2e76). Thanks!

**Review by sarayourfriend — APPROVED** [review]
Needs a rebase for the package-lock but tested locally and poked around the app and it LGTM!

File: .github/workflows/ci.yml
Comment (zackkrida) [inline .github/workflows/ci.yml:23]: Thanks for this!

**Review by zackkrida — APPROVED** [review]
LGTM! Just need the `package-lock` conflicts fixed.

**Pull request merged**
